cmake/CPM.cmake

```diff
@@ -544,6 +627,15 @@ function(CPMAddPackage)
       set(${CPM_ARGS_NAME}_BINARY_DIR ${CPM_FETCHCONTENT_BASE_DIR}/${lower_case_name}-build)
       set(${CPM_ARGS_NAME}_ADDED YES)
       set(${CPM_ARGS_NAME}_SOURCE_DIR ${download_directory})
+
+      if(DEFINED CPM_ARGS_GIT_TAG)
```

Comment:
This check implies that the download_directory passed to cpm_check_working_dir_is_clean is a git repo.
This in turn should remove the need to check for the edgecase of a download_directory in an unrelated parent git repo.

In this commit the check is still present though.

Reply:
That's a good point, we can probably safely remove the check. Lmk if you want to remove it in this PR.